helix-term/src/commands/typed.rs

```diff
@@ -2483,6 +2483,101 @@ fn move_buffer(
     Ok(())
 }
 
+fn trim_whitespace_impl(doc: &Document, selection: &Selection) -> Transaction {
+    /// Maps reverse index to non-reverse index
+    fn map_reverse(len: usize, rev: usize) -> usize {
```

Comment:
This looks like it could be a useful utility/helper somewhere else. Why was it defined locally? Is it to make the `fn` easier to call, as `len: usize` is more convenient than `len: NonZero<usize>`?